ethcore/src/state/account.rs

```diff
@@ -154,12 +162,17 @@ impl Account {
 
 	/// Create a new contract account.
 	/// NOTE: make sure you use `init_code` on this before `commit`ing.
-	pub fn new_contract(balance: U256, nonce: U256) -> Account {
+	pub fn new_contract(balance: U256, nonce: U256, original_storage_root: H256) -> Account {
```

Comment:
I may have a wrong impression about the existing code (I noted that yesterday evening), still likes Option version better, but it is only a suggestion.

Reply:
I think it may actually be more confusing. In account model, we currently distinguish:

* Account that does not exist. (`None`)
* Account exists, and storage is empty. (`Some(KECCAK_NULL_RLP)`)
* Account exists, and storage is not empty. (`Some(x)` where `x != KECCAK_NULL_RLP`)

For the purpose of `original_storage_root`, we don't need to distinguish `None` and `Some(KECCAK_NULL_RLP)`. So without `Option` it may work better here.

Reply:
I guess it is good to keep it in mind, could be a point to consider in #9427 (Option<Option< is ugly but there is other ways).